lib/src/provider/asset_picker_provider.dart

```diff
@@ -479,9 +483,17 @@ class DefaultAssetPickerProvider
         (PathWrapper<AssetPathEntity> p) => p.path == path.path,
       );
       if (index != -1) {
-        _paths[index] = _paths[index].copyWith(thumbnailData: data);
+        _paths[index] =
+            _paths[index].copyWith(thumbnailData: data, assetCount: assetCount);
         notifyListeners();
       }
+      // update _currentPath fix: https://github.com/fluttercandies/flutter_wechat_assets_picker/issues/602
+      if (index == 0) {
+        _currentPath = _currentPath?.copyWith(
+          thumbnailData: data,
+          assetCount: assetCount,
+        );
+      }
```

Comment:
You don't need to copy each path here, no?

Reply:
What I mean is you can update the `_currentPath` only since others will be updated somewhere else.

Reply:
In my testing, only updating `_currentPath` does not solve the problem. 😔